writer/writer.go

```diff
@@ -460,14 +469,32 @@ func (pw *ParquetWriter) Flush(flag bool) error {
 						nullCount = page.Header.DataPageHeaderV2.Statistics.NullCount
 					}
 
-					columnIndex.MinValues[l] = minVal
-					columnIndex.MaxValues[l] = maxVal
+					// Handle nil values properly in column index.
+					// As per Parquet spec, min/max values are for non-null values only.
+					// For null values, we set min/max values to empty byte arrays and mark the page as potentially containing nulls.
+					if minVal == nil || maxVal == nil {
+						columnIndex.MinValues[dataPageIndex] = []byte{}
+						columnIndex.MaxValues[dataPageIndex] = []byte{}
+						columnIndex.NullPages[dataPageIndex] = false
+						// Check if this is actually a null page (all values are null)
+						if nullCount != nil && page.Header.DataPageHeader != nil {
+							totalValues := page.Header.DataPageHeader.NumValues
+							if *nullCount == int64(totalValues) {
+								columnIndex.NullPages[dataPageIndex] = true
+							}
+						}
```

Comment:
If either min/macVal is nil, ideally we do not need to check if all values are nil.